### ISSUE TITLE
Cannot serialize numpy.float32

### ISSUE THREAD

**Issue by stanton119** [facebook/prophet#1824]
Hi,

I have an issue serialising a model that was fit with a column of type float32.

Version:
fbprophet == 0.7.1
pystan == 2.19.1.1

Example code:
```
import pandas as pd
import numpy as np
from fbprophet import Prophet
from fbprophet.serialize import model_to_json

df = pd.DataFrame()
df["ds"] = pd.date_range(start="2020-01-01", end="2021-01-01", freq="1MS")
df["y"] = np.random.rand(df.shape[0])
df["y"] = df["y"].astype("float32")

model = Prophet()
model.fit(df)
model_json = model_to_json(model)
```

I did a bit of investigation:
```
# after returning the model_json before serialization:
import json
for key in model_json:
    try:
        json.dumps(model_json[key])
    except TypeError:
        print(key)
```

This returns only `y_scale` and `type(model_json["y_scale"]) => numpy.float32`.

Within serialize.py `y_scale` is in the list `SIMPLE_ATTRIBUTES`, as such it is serialised directly rather than using np.to_list()

This may relate to pull request #1656 and issue #1701 

Thanks!

**Comment by bletham** [comment]
Sorry for the slow reply on this! Thanks for providing replication code and figuring out y_scale as the issue.

I'm actually not able to reproduce the issue still, which I'm thinking now might be due to a difference in numpy version? Or python version? I'm not quite sure. Can you give your versions of numpy and python? I have numpy 1.19.1 and python 3.8.3.

This is the full code that works for me but which I expect to error based on your report:
```Python
import pandas as pd
import numpy as np
from fbprophet import Prophet
from fbprophet.serialize import model_to_json, model_from_json

df = pd.DataFrame()
df["ds"] = pd.date_range(start="2020-01-01", end="2021-01-01", freq="1MS")
df["y"] = np.random.rand(df.shape[0])
df["y"] = df["y"].astype("float32")

model = Prophet()
model.fit(df)
model_json = model_to_json(model)
model_new = model_from_json(model_json)  # should error?
```

**Comment by stanton119** [comment]
Maybe it's an edge case with a specific version set... This was using:
Numpy: 1.19.5
Python 3.8.6

**Comment by bletham** [comment]
I think we're making some progress - the issue might be in pandas actually.

This produces an error for me, since numpy float32s don't serialize:
```Python
import numpy as np
import json

a = np.float32(4.0)
json.dumps(a) # Error!
```

But the code I posted above does not. It looks like the reason is because when I fit the model, `model.y_scale` is a regular float and not a np.float32. In particular:
```Python
import pandas as pd
import json

df = pd.DataFrame()
df["ds"] = pd.date_range(start="2020-01-01", end="2021-01-01", freq="1MS")
df["y"] = np.random.rand(df.shape[0])
df["y"] = df["y"].astype("float32")

print(type(df['y'][0]))  # <class 'numpy.float32'>

y_scale = df["y"].max()
print(type(y_scale))  # <class 'float'> , NOT np.float32

json.dumps(df['y'][0])  # Errors
json.dumps(y_scale)  # Works
```
So it looks like for me, `pd.max` is producing a float when applied to a column of np.float32s, whereas for you (and other people that have reported this issue) it is remaining a np.float32.

What version of pandas are you using? I got the result above (pd.max is float) with 1.1.4 and 1.2.3.

And could you run the code snippet above to confirm that indeed pd.max is producing np.float32 for you?

**Comment by stanton119** [comment]
Yes! that's the problem. I'm on Pandas 1.2.0 and I'm getting the .max() as a numpy type:
```
y_scale = df["y"].max()
print(type(y_scale))  # <class 'numpy.float32'>
```

I think you're commit will solve it.

**Comment by bletham** [comment]
Awesome, thanks for your help in figuring out what was going on here!

**Comment by bletham** [comment]
This fix has been pushed to pypi in the new v1.0 release.
